src/gfx/shaders/AO.frag

```diff
@@ -50,7 +50,11 @@ void main() {
   // remap coordinates to prevent noise exture rescale
   vec2 vUvNoise = vUv / srcTexelSize * noiseTexelSize;
   //[0, 1] -> [-1, 1]
-  vec3 normal = (texture2D(normalTexture, vUv).rgb * 2.0 - 1.0);
+  vec4 normalData = texture2D(normalTexture, vUv);
+  vec3 normal = (normalData.rgb * 2.0 - 1.0);
```

Comment:
Here you use the value from a-channel as a frontFacing boolean value. But it is not a boolean, it's a normal multiplier (see the code for storage).

Reply:
I partly disagree with you. I use it in valid way as float number, but now I know how to do it more correct and more understandable (using something like check on equality to zero value for float values)
I add small change in this code fragment

Reply:
Oh, there is a misunderstanding. I've said 'boolean', meant 'frontFace' flag. There is no question about the variable type. 
I'm sorry about that.